### PR TITLE
Remove reference to CMS GC

### PR DESCRIPTION
**Pull request by jayeff**
CMS was removed with JDK 14 https://openjdk.java.net/jeps/363

## Summary of the changes / Why this improves CrateDB

As noted in https://github.com/crate/crate/pull/11212#pullrequestreview-629984161 CMS is no longer part of current JDK.

This PR removes documentation on how to enable it.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/config/environment.rst [hunk 1]
```diff
@@ -118,40 +118,10 @@ General
...
-Before CrateDB 4.1 it defaulted to use the `Concurrent Mark Sweep` garbage
-collector. If you'd like to continue using CMS, you can switch setting the
-following :ref:`CRATE_JAVA_OPTS <conf-env-java-opts>`::
-
-
-  export CRATE_JAVA_OPTS="-XX:-UseG1GC -XX:+UseCMSInitiatingOccupancyOnly -XX:+UseConcMarkSweepGC"
-
-
-Logging
-~~~~~~~
```
Comment (jayeff) [inline docs/config/environment.rst:140]: I removed the GC Logging section as well as this is not really related to env variables

**Review by mfussenegger — APPROVED** [review]
Nice :+1:

**Review by amotl — APPROVED** [review]

**Comment by jayeff** [comment]
@Mergifyio backport v/4.5

**Comment by jayeff** [comment]
@Mergifyio backport 4.5
